smack-experimental/src/main/java/org/jivesoftware/smackx/httpfileupload/HttpUploadException.java

```diff
@@ -0,0 +1,106 @@
+/**
+ *
+ * Copyright the original author or authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jivesoftware.smackx.httpfileupload;
+
+import java.io.IOException;
+import java.net.URL;
+
+import org.jivesoftware.smackx.httpfileupload.element.Slot;
+
+
+/**
+ * An exception class to provide additional information in case of exceptions during file uploading.
+ *
+ */
+public abstract class HttpUploadException extends IOException {
+
+    private static final long serialVersionUID = 1L;
+
+    protected HttpUploadException(String message) {
+        super(message);
+    }
+
+    protected HttpUploadException(String message, Throwable wrappedThrowable) {
+        super(message, wrappedThrowable);
+    }
+
+
+    /**
+     * Exception thrown when http response returned after upload is not 200.
+     */
+    public static class UnsuccessfulUploadException extends HttpUploadException {
+
+
+        private static final long serialVersionUID = 8494356028399474995L;
+        private final int httpStatus;
+        private final String responseMsg;
+        private final long fileSize;
+        private final URL putUrl;
+
+        public UnsuccessfulUploadException(int httpStatus, String responseMsg, long fileSize, URL putUrl) {
+            super(String.format("Error response %d from server during file upload %s, file size: %d, put Url: %s",
```

Comment:
Note that this is a situation where you typically would not use `String.format()`, as you do not use any format specifiers. String concatenation using the `+` operator would just work fine here, would be more readable and more Java-idomatic.